app/js/checkAnswers.js

```diff
@@ -129,3 +129,83 @@ function displayResult(earnedPoints, earnedPercentage, answeredQuestions) {
     document.querySelector(".answered_questions").innerHTML = answeredQuestions
     document.querySelector(".score_percentage").innerHTML = earnedPercentage
 }
+
+/**
+ * function adds event listeners to .question and listens for click event within here
+ * it then updates the class of the span containing the question number allowing styling to be applied
+ *
+ */
+function addAnswerEventListeners() {
+    document.querySelectorAll('.question').forEach(function (input) {
+        input.addEventListener('click', function(e) {
+            if (e.target.tagName == 'INPUT') {
+                let nav = document.querySelector('#question-nav')
+                let id = parseInt(this.dataset['id']) - 1
+                nav.children[id].classList.add('answered-nav-box')
+            }
+        })
+    })
+}
+
+/**
+ * function removes current status from all questions and then adds current status
+ * to the current question allowing styling to be applied
+ *
+ * @param id is the id of the active question
+ *
+ */
+function trackActiveQuestion(id) {
+    let nav = document.querySelector('#question-nav')
+    for (let i = 0; i < nav.children.length; i++) {
+        nav.children[i].classList.remove('current-nav-box')
+    }
+    nav.children[id - 1].classList.add('current-nav-box')
+}
+
+/*
+ *  this get the unanswered questions and puts their question id into an array
+ * @returns the array of question ids that havent been answered
+ */
+function questionAnswered() {
+    const qAmount = document.querySelectorAll('.question').length
+    let answers = getUserAnswers(qAmount)
+    let answersArr = Object.values(answers)
+    let unanswered = []
+    answersArr.forEach(function (value, qid) {
+        qid++
+        if(value == 'unanswered') {
+            unanswered.push(qid)
+        }
+    })
+    if (unanswered.length == 0) {
```

Comment:
why not just return `unanswered.length` here. when you use the array on line 9 you grab the length anyway

Reply:
refactored in 89e3e12